permission-portal-frontend/client/src/store/index.js

```diff
@@ -15,95 +15,62 @@ const createStore = (WrappedComponent) => {
       this.authStateListener = auth.onAuthStateChanged(async (user) => {
         if (user) {
           console.log('User signed in')
-          // signed in, get user's document from the db
-          const userDocumentSnapshot = await db.collection('users').doc(user.email).get()
-          // update the store with data from this document and set isSignedIn to true
-          rootStore.user.__update({ ...userDocumentSnapshot.data(), email: userDocumentSnapshot.id, isSignedIn: true })
-          // set up a listener to respond to current user's document changes
+          // get user's document from the db and at the same time set up a listener to respond to document changes
           if (this.userDocumentListener === null) {
             this.userDocumentListener = db
               .collection('users')
               .doc(user.email)
               .onSnapshot((updatedUserDocumentSnapshot) => {
-                console.log('Remote user document changed')
+                // update the store with data from this document and set isSignedIn to true
                 rootStore.user.__update({
                   ...updatedUserDocumentSnapshot.data(),
                   email: updatedUserDocumentSnapshot.id,
                   isSignedIn: true,
                 })
+                // If DNE, set up organization listener within this callback,
+                // since it relies on rootStore.user.organizationID being set
+                if (this.organizationDocumentListener === null) {
+                  this.organizationDocumentListener = db
+                    .collection('organizations')
+                    .doc(rootStore.user.organizationID)
+                    .onSnapshot((updatedOrganizationDocumentSnapshot) => {
+                      console.log('Remote organization document changed')
+                      rootStore.organization.__update({
+                        ...updatedOrganizationDocumentSnapshot.data(),
+                        id: updatedOrganizationDocumentSnapshot.id,
+                      })
+                      if (rootStore.user.isAdmin && this.organizationMembersListener === null) {
+                        this.organizationMembersListener = db
+                          .collection('users')
+                          .where('organizationID', '==', rootStore.organization.id)
+                          .onSnapshot((updatedUsersSnapshot) => {
+                            rootStore.organization.__setMembers(
+                              // See https://stackoverflow.com/a/24806827
+                              updatedUsersSnapshot.docs.reduce((result, userDoc) => {
```

Comment:
Could be more efficient to listen for changes here (for example, see [this code sample](https://firebase.google.com/docs/firestore/query-data/listen#view_changes_between_snapshots))

Reply:
Almost certainly is more efficient. Added an issue for it, though we should wait until pagination has been sorted out to implement: https://github.com/covid19risk/permission-portal/issues/199